scripts/bump_versions.py

```
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+"""Bump version of selected packages or core requirements (JupyterLab)"""
+import sys
+from dataclasses import dataclass
+from pathlib import Path
+from typing import List
+
+from integrity import CHANGELOG, PIPE_FILE as PIPELINE
+
+ROOT = Path.cwd()
+
+sys.path.insert(0, str(ROOT))
+
+if True:
+    # a workaround for isort 4.0 limitations
+    # see https://github.com/timothycrosley/isort/issues/468
+    from versions import (  # noqa
+        JUPYTER_LSP_VERSION,
+        JUPYTERLAB_LSP_VERSION,
+        JUPYTERLAB_VERSION,
+        REQUIRED_JUPYTERLAB,
+    )
+
+
+META_PACKAGE = Path("packages/metapackage/package.json")
+JUPYTERLAB_LSP_PACKAGE = Path("packages/jupyterlab-lsp/package.json")
+README = Path("README.md")
+
+NPM_PACKAGE_VERSION_TEMPLATE = '"version": "{version}"'
+
+
+@dataclass
+class VersionLocation:
+    path: Path
+    template: str
+
+
+@dataclass
+class PackageVersionInfo:
+    name: str
+    current_version: str
+    locations: List[VersionLocation]
+
+    def maybe_change_version(self, dry: bool):
+        print(f"Current {self.name} version is: {self.current_version}")
+        version = input("Change it to [default=skip]: ").strip()
+        if version:
+            self.change_version(new_version=version, dry=dry)
+
+    def change_version(self, new_version: str, dry: bool):
+
+        changelog = CHANGELOG.read_text()
+        assert new_version in changelog
```

should raise a verbose exception